pkg/kubectl/cmd/set/set_serviceaccount.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package set
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/util/i18n"
+)
+
+var (
+	serviceaccountResources = `
+	replicationcontroller (rc), deployment (deploy), daemonset (ds), job, replicaset (rs), statefulset`
+
+	serviceaccountLong = templates.LongDesc(`
+	Update ServiceAccount of pod template resources.
+
+	Possible resources (case insensitive) can be: 
+	` + serviceaccountResources)
+
+	serviceaccountExample = templates.Examples(`
+	# Set Deployment nginx-deployment's ServiceAccount to serviceaccount1
+	$kubectl set service-account deployment nginx-deployment serviceaccount1
+
+	# Short form of the above command
+	$kubectl set sa deploy nginx-deployment serviceaccount1
+
+	# Print result in yaml format of updated nginx deployment from local file, without hitting apiserver
+	$kubectl set sa -f nginx-deployment.yaml serviceaccount1 --local --dry-run -o yaml
+	`)
+)
+
+// serviceAccountConfig encapsulates the data required to perform the operation.
+type serviceAccountConfig struct {
+	fileNameOptions resource.FilenameOptions
+
+	mapper                 meta.RESTMapper
+	encoder                runtime.Encoder
+	decoder                runtime.Decoder
+	out                    io.Writer
+	err                    io.Writer
+	dryRun                 bool
+	shortOutput            bool
+	all                    bool
+	record                 bool
+	output                 string
+	changeCause            string
+	local                  bool
+	categoryExpander       resource.CategoryExpander
+	clientMapper           resource.ClientMapper
+	typer                  runtime.ObjectTyper
+	namespace              string
+	enforceNamespace       bool
+	args                   []string
+	_print                 func(obj runtime.Object) error
```

Comment:
ah, I saw the previous comments _print  is use to avoid keyword, how about `saprint`  or sth else, @deads2k wdyt  ?